server/connection.ts

```diff
@@ -0,0 +1,280 @@
+import { Socket } from 'socket.io';
```

Comment:
This file is hard to diff due to both including the refactor into Connection and the new code. I'd suggest applying those changes in separate PRs.